larray/tests/test_array.py

```diff
@@ -61,6 +72,56 @@ def test_slice_strings(self):
         self.assertEqual(_to_key(':'), slice(None))
 
 
+# TODO : remove specific tests for Python 2 version of Metadata once Python 2.7 version of LArray will be dropped
+def test_array_metadata(attrs, tmpdir):
+    # __eq__ (for Python 2 version)
+    attrs2 = attrs.copy()
+    print(type(attrs), type(attrs2))
+    assert attrs2 == attrs
+
+    # set/get metadata to/from an array
+    arr = ndtest((3, 3))
+    arr.attrs = attrs
+    assert arr.attrs == attrs
+
+    # access item
+    assert arr.attrs.date == attrs.date
+
+    # add new item
+    arr.attrs.city = 'London'
+    assert arr.attrs.city == 'London'
+
+    # update item
+    arr.attrs.city = 'Berlin'
+    assert arr.attrs.city == 'Berlin'
+
+    # __contains__ (for Python 2 version)
+    assert 'city' in arr.attrs
+
+    # delete item
+    del arr.attrs.city
+    assert arr.attrs == attrs
+
+    # __reduce__ and __reduce_ex__ (for Python 2 version)
+    import pickle
+    fname = os.path.join(tmpdir.strpath, 'test_metadata.pkl')
+    with open(fname, 'wb') as f:
+        pickle.dump(attrs, f)
+    with open(fname, 'rb') as f:
+        attrs2 = Metadata(OrderedDict(pickle.load(f)))
+    assert attrs2 == attrs
+
+
+def test_metadata_hdf(attrs, tmpdir):
+    key = 'meta'
+    fname = os.path.join(tmpdir.strpath, 'test_metadata.hdf')
+    with LHDFStore(fname) as store:
+        ndtest(3).to_hdf(store, key)
```

Comment:
Because `Metadata.to_hdf` uses the standard attribute [`attrs`](https://www.pytables.org/usersguide/tutorials.html#setting-and-getting-user-attributes) associated to each node object. 
So, I first need to store an arbitrary object and then call 
```python 
  return hdfstore.get_storer(key).attrs.metadata
``` as in `Metadata.to_hdf`.